nipype/pipeline/plugins/multiproc.py

```diff
@@ -124,10 +135,60 @@ def __init__(self, plugin_args=None):
         self.memory_gb = self.plugin_args.get('memory_gb',  # Allocate 90% of system memory
                                               get_system_total_memory_gb() * 0.9)
         self.raise_insufficient = self.plugin_args.get('raise_insufficient', True)
-
+        
+        self.n_gpus_visible = self.gpu_count()
+        self.n_gpus = self.plugin_args.get('n_gpus', self.n_gpus_visible)
+        self.n_gpu_proc = self.plugin_args.get('ngpuproc', 1)
+        
+        # Check plugin args
+        if self.plugin_args:
+            if 'non_daemon' in self.plugin_args:
+                non_daemon = plugin_args['non_daemon']
+            if 'n_procs' in self.plugin_args:
+                self.processors = self.plugin_args['n_procs']
+            if 'memory_gb' in self.plugin_args:
+                self.memory_gb = self.plugin_args['memory_gb']
+            if 'n_gpus' in self.plugin_args:
+                self.n_gpus = self.plugin_args['n_gpus']
+            if 'ngpuproc' in self.plugin_args:
+                self.n_gpu_proc = self.plugin_args['ngpuproc']
+                
+                
+        #total no. of processes allowed on all gpus
+        if self.n_gpus > self.n_gpus_visible:
+            logger.info('Total number of GPUs (%d) requested exceeds available number of GPUs (%d). Using all %d GPU(s).'%(self.n_gpus,self.n_gpus_visible, self.n_gpus_visible))
+            self.n_gpus = self.n_gpus_visible
+            self.total_gpu_processors = self.n_gpus * self.n_gpu_proc
+        else:
+            #total gpu_processors = no.of GPUs * no.of threads per single GPU
+            self.total_gpu_processors = self.n_gpus * self.n_gpu_proc
+            
+
+        #form a GPU queue first
+        gpus=[]
+        try:
```

Comment:
this try catch is similar to gpu_count below. perhaps can be refactored to a common function.